src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -200,12 +203,55 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
             // OtpUserController. Therefore, the request should be limited to return just the entity matching the
             // requesting user.
             return persistence.getResponseList(Filters.eq("_id", requestingUser.otpUser.id), offset, limit);
+        } else if (requestingUser.isThirdPartyUser()) {
+            // A user id must be provided if the request is being made by a third party user.
+            JsonUtils.logMessageAndHalt(req,
+                HttpStatus.BAD_REQUEST_400,
+                String.format("The parameter name (%s) must be provided.", USER_ID_PARAM));
+            return null;
         } else {
             // For all other cases the assumption is that the request is being made by an Otp user and the requested
             // entities have a 'userId' parameter. Only entities that match the requesting user id are returned.
-            return persistence.getResponseList(Filters.eq("userId", requestingUser.otpUser.id), offset, limit);
+            return persistence.getResponseList(Filters.eq(USER_ID_PARAM, requestingUser.otpUser.id), offset, limit);
         }
     }
+//        private ResponseList<T> getMany(Request req, Response res) {
```

Comment:
Remove commented out block if it's no longer needed.

Reply:
Updated.